src/main/scala/wdltool/graph/WomGraph.scala

```diff
@@ -39,42 +41,102 @@ class WomGraph(graphName: String, graph: Graph) {
+        |}
+      """.stripMargin
+    case _ => (graphNode.outputPorts ++ graphNode.inputPorts) map portLine mkString "\n  "
   }
 
   /**
     * For most graph nodes, we draw a single box for the node, containing the name and input and output ports.
     * For each input/output port, we include the links to upstream ports.
     */
-  private def defaultNodesAndLinks(graphNode: GraphNode): NodesAndLinks = {
-    val clusterName = "cluster_" + clusterCount.getAndIncrement()
-    val portNodes = (graphNode.outputPorts ++ graphNode.inputPorts) map { p =>
-      s"${p.graphId} [shape=${p.graphShape} label=${p.graphName}];"
-    }
-    val links = for {
-      inputPort <- graphNode.inputPorts
-      upstreamPort = inputPort.upstream
-    } yield s"${upstreamPort.graphId} -> ${inputPort.graphId}"
+  private def nodesAndLinks(graphNode: GraphNode): NodesAndLinks = {
+    val internals = internalNodesAndLinks(graphNode)
+    val clusterName = nextCluster
 
     val singleNode =
       s"""
          |subgraph $clusterName {
          |  style=filled;
          |  fillcolor=${graphNode.graphFillColor};
-         |  ${portNodes.mkString(sep="\n  ")}
+         |  ${portLines(graphNode)}
+         |  ${internals.nodes.mkString(System.lineSeparator + "  ")}
          |  ${graphNode.graphId} [shape=plaintext label=${graphNode.graphName}]
          |}
          |""".stripMargin
 
-    NodesAndLinks(Set(singleNode), links)
+    NodesAndLinks(Set(singleNode), internals.links ++ upstreamLinks(graphNode))
+  }
+
+  private def internalNodesAndLinks(graphNode: GraphNode): NodesAndLinks = graphNode match {
+    case scatter: ScatterNode => internalScatterNodesAndLinks(scatter)
+    case _ => NodesAndLinks.empty
   }
+
+  def internalScatterNodesAndLinks(scatter: ScatterNode) = {
+    val innerGraph = listAllGraphNodes(scatter.innerGraph) wrapNodes { n =>
+      s"""
+         |subgraph $nextCluster {
+         |  style=filled;
+         |  fillcolor=lightgray;
+         |  ${n.mkString(sep = "\n  ")}
+         |}
+         |""".stripMargin
+    }
+    val scatterVariableNodesAndLinks = {
+      val scatterVariableExpressionId = s""""${UUID.randomUUID}""""
+      val scatterVariableClusterName = nextCluster
+      val node = s"""
+         |subgraph $scatterVariableClusterName {
+         |  style=filled;
+         |  fillcolor=${scatter.graphFillColor};
+         |  ${scatter.scatterVariableMapping.scatterInstantiatedExpression.inputPorts map portLine mkString "\n  "}
+         |  $scatterVariableExpressionId [shape=plaintext label="${scatter.scatterVariableMapping.graphInputNode.name} in ..."]
+         |}
+         """.stripMargin
+
+      val scatterVariableInputLink = s"$scatterVariableExpressionId -> ${scatter.scatterVariableMapping.graphInputNode.singleOutputPort.graphId} [style=dashed ltail=$scatterVariableClusterName arrowhead=empty]"
+      val outputLinks = scatter.outputMapping map { case outputPort @ ScatterGathererPort(_, _, outputToGather, _) =>
```

Comment:
probably just capturing would suffice, the pattern match seems unnecessary.  might even fit on one line.  😄